c-c++.md

```diff
@@ -4,19 +4,32 @@
 To build code with the optimal processor features use the following. If you want to support both Graviton
 and Graviton2 you'll have to limit yourself to the Graviton features.
 
-CPU      | GCC                  | LLVM
----------|----------------------|-------------
-Graviton | `-march=armv8-a+crc+crypto` | `-march=armv8-a+crc+crypto`
+CPU       | GCC                         | LLVM
+----------|-----------------------------|-------------
+Graviton  | `-march=armv8-a+crc+crypto` | `-march=armv8-a+crc+crypto`
 Graviton2 | `-march=armv8.2-a+fp16+rcpc+dotprod+crypto` |`-march=armv8.2-a+fp16+rcpc+dotprod+crypto`
+Graviton3 | `-march=armv8.2-a+fp16+rcpc+dotprod+crypto+sve+bf16+pauth+rng` |`-march=armv8.2-a+fp16+rcpc+dotprod+crypto+sve+bf16+pauth+rand`
```

Comment:
i don't know that we want to have +sve here, it's just going to end up generating sve code when it might not be optimal. pauth+rng+bf16 aren't going to be emitted from the compiler generally either. We should probably just say use the Grv2 flags.

Reply:
`+sve` does not enable auto-vectorization for SVE.
To auto-generate SVE instructions, one needs to use `--param=aarch64-autovec-preference=2`, `3`, or `4`:

```
@item aarch64-autovec-preference
Force an ISA selection strategy for auto-vectorization.  Accepts values from
0 to 4, inclusive.
@table @samp
@item 0
Use the default heuristics.
@item 1
Use only Advanced SIMD for auto-vectorization.
@item 2
Use only SVE for auto-vectorization.
@item 3
Use both Advanced SIMD and SVE.  Prefer Advanced SIMD when the costs are
deemed equal.
@item 4
Use both Advanced SIMD and SVE.  Prefer SVE when the costs are deemed equal.
@end table
The default value is 0.
```

With `+sve`, if the code contains SVE intrinsics the compiler will emit SVE instructions.
Same for `+rng` and `+bf16`, even if the compiler is not able to automatically emit those instructions, code with intrinsics need the flags set to enable the compiler intrinsic call and the assembly.